src/conduits/RwaInputConduit3.sol

```diff
@@ -211,14 +211,44 @@ contract RwaInputConduit3 {
         emit Push(to, daiBalance);
     }
 
+    /**
+     * @notice Method to swap specific amount of USDC contract balance to DAI through PSM and push it into RwaUrn address.
+     * @dev `msg.sender` must first receive push acess through mate().
+     * @param amt Gem amount
+     */
+    function push(uint256 amt) external isMate {
+        uint256 gemBalance = gem.balanceOf(address(this));
+        require(gemBalance >= amt, "RwaInputConduit3/not-enough-gems");
+
+        psm.sellGem(address(this), amt);
+
+        uint256 daiBalance = dai.balanceOf(address(this));
+        dai.transfer(to, daiBalance);
+
+        emit Push(to, daiBalance);
+    }
+
```

Comment:
There's some significant code duplication here. Maybe extract the `push` logic into an internal method and have the public `push()` and `push(wad)` to call that implementation:

```solidity
function push() external isMate {
    _doPush(dai.balanceOf(address(this)));
}

function push(uint256 wad) external isMate {
    _doPush(wad);
}

function _doPush(uint256 wad) internal {
    // ...
}
```

Reply:
Done